### PR TITLE
LPS-113561 Avoid use of Liferay.provide in portal-settings-authentication-cas-web

### PR DESCRIPTION
**Pull request by julien**
As part of [LPS-113561](https://issues.liferay.com/browse/LPS-113561) we eventually want to deprecate the `Liferay.provide` function

To test this change:

- From the "Global Menu" select the "Control Panel" tab
- Click on "Configuration > Instance Settings"
- In the "Security" section click on "SSO"
- Click on the kebab menu to the right of "CAS Server" and click on "Test CAS Configuration"

As a result a modal window should appear and depending on the values you entered for your CAS server it should either notify you that the connection was successful or that it failed and no JS errors should appear in your browser's console

### CODE REVIEW

**Comment by julien** [comment]
ci:test:sf

**Comment by julien** [comment]
ci:test:relevant

**Comment by julien** [comment]
In this case, what I've done to test locally until we decide on something is the same thing I mentioned [here](https://github.com/wincent/liferay-portal/pull/343#issuecomment-642485580) 

```jsp
<aui:script>
	window['<portlet:namespace />testCasSettings'] = function () {
		var A = AUI();

		A.use('aui-io-plugin-deprecated', 'liferay-util-window', function() {
			var data = {};
                         ...
                });
        };
</aui:script>
```

**Comment by jbalsas** [comment]
We don't we move this to `openModal` and get rid of those dependencies? 

/cc @markocikos

**Comment by julien** [comment]
So about `openModal`:

It's pretty easy to use, but [this](https://github.com/liferay/liferay-portal/blob/2a9b1908a5a336d7ae1e5096e2c012f5e19d2e5a/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/Modal.js#L405) is throwing an error - so I've changed that 

In my case the URL that's used for the modal is 

```
http://localhost:8080/group/control_panel/manage?p_p_id=com_liferay_configuration_admin_web_portlet_InstanceSettingsPortlet&p_p_lifecycle=0&p_p_state=exclusive&p_p_mode=view&_com_liferay_configuration_admin_web_portlet_InstanceSettingsPortlet_mvcRenderCommandName=%2Fportal_settings%2Ftest_opensso&_com_liferay_configuration_admin_web_portlet_InstanceSettingsPortlet_openSsoLoginURL=http%3A%2F%2Fopenssohost.example.com%3A8080%2Fopensso%2FUI%2FLogin%3Fgoto%3Dhttp%3A%2F%2Fportalhost.example.com%3A8080%2Fc%2Fportal%2Flogin&_com_liferay_configuration_admin_web_portlet_InstanceSettingsPortlet_openSsoLogoutURL=http%3A%2F%2Fopenssohost.example.com%3A8080%2Fopensso%2FUI%2FLogout%3Fgoto%3Dhttp%3A%2F%2Fportalhost.example.com%3A8080%2Fweb%2Fguest%2Fhome&_com_liferay_configuration_admin_web_portlet_InstanceSettingsPortlet_openSsoServiceURL=http%3A%2F%2Fopenssohost.example.com%3A8080%2Fopensso&_com_liferay_configuration_admin_web_portlet_InstanceSettingsPortlet_openSsoScreenNameAttr=uid&_com_liferay_configuration_admin_web_portlet_InstanceSettingsPortlet_openSsoEmailAddressAttr=mail&_com_liferay_configuration_admin_web_portlet_InstanceSettingsPortlet_openSsoFirstNameAttr=givenName&_com_liferay_configuration_admin_web_portlet_InstanceSettingsPortlet_openSsoLastNameAttr=sn&_com_liferay_configuration_admin_web_portlet_InstanceSettingsPortlet_bodyCssClass=dialog-iframe-popup
```

Another thing I'm failing to do for now is getting the same css styles as we had with `liferay-util-window`.

**Comment by jbalsas** [comment]
> It's pretty easy to use, but this is throwing an error - so I've changed that

You shouldn't, at least not for this. You have an `EXCLUSIVE` url. that's a url that only returns you some html, and not the whole portal. That's why you don't get the styles or the JS scripts.

`A.Plugin.IO` is the plugin that makes sure to evaluate scripts when injected in a page.

So, without testing, you should be doing `Liferay.Util.fetch()` and use that as the modal body passing it through `globalEval` (or the fragment mechanism of our choice).

**Comment by julien** [comment]
@jbalsas got it, I first thought I should "request" the page via fetch (because `A.Plugin.IO` looks like an "ajax" wrapper) but I wasn't sure. Thanks

**Comment by julien** [comment]
Here's a screenshot of how it looked before:

before

And how it looks after:

after

**Comment by julien** [comment]
Here's a screenshot of how it looked before:

cas-before

And how it looks after:

cas-after

File: modules/apps/portal-settings/portal-settings-authentication-cas-web/src/main/resources/META-INF/resources/dynamic_include/com.liferay.portal.settings.web/cas.jsp [hunk 1]
```diff
@@ -80,19 +78,40 @@ String noSuchUserRedirectURL = casConfiguration.noSuchUserRedirectURL();
...
 		});
+
+		Liferay.Util.fetch(url)
+			.then(function (response) {
+				return response.text();
+			})
+			.then(function (text) {
+				var parser = new DOMParser();
+
+				var doc = parser.parseFromString(text, 'text/html');
```
Comment (wincent) [inline modules/apps/portal-settings/portal-settings-authentication-cas-web/src/main/resources/META-INF/resources/dynamic_include/com.liferay.portal.settings.web/cas.jsp:99]: Why do you need to parse the HTML here? It already is HTML.

**Comment by wincent** [comment]
> So, without testing, you should be doing `Liferay.Util.fetch()` and use that as the modal body passing it through `globalEval` (or the fragment mechanism of our choice).

About evaluating JS scripts in the body, I believe [this code](https://github.com/liferay/liferay-portal/blob/2a9b1908a5a336d7ae1e5096e2c012f5e19d2e5a/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/Modal.js#L205-L219) already handles that for you, so no need to manually `globalEval` or use any other trick.

File: modules/apps/portal-settings/portal-settings-authentication-cas-web/src/main/resources/META-INF/resources/dynamic_include/com.liferay.portal.settings.web/cas.jsp (hunk 1 above)
Comment (julien) [reply modules/apps/portal-settings/portal-settings-authentication-cas-web/src/main/resources/META-INF/resources/dynamic_include/com.liferay.portal.settings.web/cas.jsp:99]: Ok, I didn't realize that. I'll udpate that and the other PRs (#343, #344) where I did exactly the same mistake.

**Comment by wincent** [comment]
Given your test plan and screenshots, I am just going to forward without testing it myself, trusting CI.

trust

**Comment by wincent** [comment]
ci:forward

**Pull request closed without merging**